contracts/PeeranhaToken.sol

```diff
@@ -65,42 +92,93 @@ contract PeeranhaToken is ERC20Upgradeable, ERC20PausableUpgradeable, ERC20Cappe
     RewardLib.WeekReward memory weekReward = peeranha.getWeekRewardContainer(period);
     
     uint256 poolToken = getPool(weekReward, period);
-    if (poolTokens[period] != poolToken) poolTokens[period] = poolToken;
+    if (poolTokens[period] == 0 && poolToken != poolTokens[period]) {
+      poolTokens[period] = poolToken;
+    }
     uint256 userReward = getUserReward(weekReward, user, period, poolToken);
```

Comment:
Move `peeranha.getWeekRewardContainer` and `getPool` inside `getUserReward` function.